src/main/java/com/clickhouse/kafka/connect/sink/db/ClickHouseWriter.java

```diff
@@ -403,18 +401,23 @@ private void doWriteCol(Record record, Column col, ClickHousePipedOutputStream s
                     break;
                 case ARRAY:
                     List<?> arrObject = (List<?>) value.getObject();
-                    int sizeArrObject = arrObject.size();
-                    BinaryStreamUtils.writeVarInt(stream, sizeArrObject);
-                    arrObject.forEach(v -> {
-                        try {
-                            if (col.getSubType().isNullable() && v != null) {
-                                BinaryStreamUtils.writeNonNull(stream);
+
+                    if (arrObject == null) {
+                        doWritePrimitive(colType, value.getFieldType(), stream, new ArrayList<>());
+                    } else {
+                        int sizeArrObject = arrObject.size();
+                        BinaryStreamUtils.writeVarInt(stream, sizeArrObject);
+                        arrObject.forEach(v -> {
+                            try {
+                                if (col.getSubType().isNullable() && v != null) {
+                                    BinaryStreamUtils.writeNonNull(stream);
+                                }
+                                doWritePrimitive(col.getSubType().getType(), value.getFieldType(), stream, v);
+                            } catch (IOException e) {
+                                throw new RuntimeException(e);
```

Comment:
Do not know what type of exceptions we can have here, but should we cases that we want to retry or put it in DLQ